src/main/java/org/antlr/intellij/plugin/configdialogs/ANTLRv4GrammarPropertiesStore.java

```diff
@@ -59,14 +60,14 @@ public ANTLRv4GrammarProperties getOrCreateGrammarProperties(String grammarFile)
 	@Nullable
 	private ANTLRv4GrammarProperties findSettingsForFile(String fileName) {
 		for (ANTLRv4GrammarProperties settings : perGrammarGenerationSettings) {
-			if (settings.fileName.equals(fileName)) {
+			WildcardFileNameMatcher wildcardFileNameMatcher = new WildcardFileNameMatcher(settings.fileName);
```

Comment:
Did you check bahavior of invalid pattern given to the constructor?

Reply:
I added something like this `"_gdsgjdwwf/\\   dfs"` and no exception has been thrown. It just didn't match any filename input i provided (as expected).

Reply:
What about invalid regexps like `[` or `*`?

Reply:
Special regex characters are escaped. Yet I can try-catch this and log warning just-in-case. What do you think?